### ISSUE TITLE
Warnings should refer to calling code location.

### ISSUE THREAD

**Issue by cvm-a** [drakegroup/sigfig#19]
The warnings in the sigfig library are reporting the location of the warning call within the sigfig library, like 
```
/usr/lib/python3.11/site-packages/sigfig/sigfig.py:586: UserWarning: warning: 3 significant figures requested from number with only 1 significant figures
```
This makes it hard to find out what was the call to sigfig.roud that actually triggered this error. Instead, if we use the stacklevel argument to warn, like
```
warn("warning: %d significant figures requested from number with only %d significant figures" % (given['sigfigs'], len(num.map)), stacklevel=2)
```
the error message is like
```
/my/calling/code.py:270: UserWarning: warning: 3 significant figures requested from number with only 1 significant figures
```

Can we add the stacklevel argument to all the warning calls?

**Comment by MikeBusuttil** [comment]
absolutely, good call.  Post a PR and it's going in!

**Comment by MikeBusuttil** [comment]
fixed in #20 & #45 and released in v1.3.17

thanks @cvm-a this is a huge improvement!  The warning behavior always made me cringe and you fixed it 🙌
